### PR TITLE
Add banner component issue #371

### PR DESCRIPTION
**Pull request by fozy81**
WIP - draft PR for issue #371 

@amyrlam I've got the basic banner template worked out. 

but...

1. I need help with Text color!...:-( I must be missing something. The css is in the `.es-banner.css` and imported via the `addon.css` file. 

please see preview: https://deploy-preview-401--ember-styleguide.netlify.app/components/banner/

2. I've allowed users to add a url not a route. In case the banner needs to point to a page outside on of the Ember website (or which ever website this component is used on). Does that sound useful or should this component be limited to only `<linkTo />` route within the website?

3. Should we offer styling options for the banner, currently just uses the brand dark color.

### CODE REVIEW

**Comment by amyrlam** [comment]
@fozy81 Sorry for the delay on my end, I'll have to wait to take a closer look when I'm back from vacation next week ~Wednesday. I'll familiarize myself with the repo then and try to help this and your other PR get moved along then.

**Comment by fozy81** [comment]
@amyrlam Enjoy your holiday, catch up some time next week. Thanks for helping.

File: addon/components/es-banner.hbs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{{#if this.show}}
+<div class="es-banner">
```
Comment (amyrlam) [inline addon/components/es-banner.hbs:2]: we'll fix the linting here to make it easier to contribute to the repo

**Comment by amyrlam** [comment]
Hey @fozy81 I went to the meeting today to get more context on what we want this component to do. Jotting down some quick rough notes of next steps. Something I can kick off...
- [ ] Investigate how the banner was implemented in `config/environment` https://github.com/ember-learn/ember-website/pull/620/files (some of the code may have been deleted on main. We may want the banner to be what that code was, look at the options there)
- [ ] See if it makes sense to bring this component into styleguide
- [ ] Sensible defaults

**Comment by fozy81** [comment]
Hi @amyrlam thanks for discussing at meeting, let me know if there's anything I can help with. :+1:

**Comment by amyrlam** [comment]
Hey @fozy81 @mansona 

>  Investigate how the banner was implemented in config/environment https://github.com/ember-learn/ember-website/pull/620/files (some of the code may have been deleted on main. We may want the banner to be what that code was, look at the options there)

Did some spelunking (thought I posted yest - whoops) and I know what @mansona meant now! The banner is added via `config/environment.js`. So the component won't be a "drop-in" per se...but it still seems like it makes sense to include in `ember-styleguide`, with instructions documented? We can build off of what @fozy81 has started

- https://github.com/ember-learn/ember-website/pull/667/files
- https://github.com/ember-learn/ember-website/pull/656/files
- https://github.com/ember-learn/ember-website/pull/620/files
- https://github.com/ember-learn/ember-website/pull/594/files

**Comment by fozy81** [comment]
@amyrlam @mansona I understand (I think) - `config/environment.js` magic! I'm bit out my depth to know the advantages of configuring the component in this way. I wonder if values for the banner component arguments could be provided by the `config/environment.js` instead? This would keep the component consistent with other styleguide components (rather than requiring a config file). For example, `<Banner @url={{this.config.link}} @message={{this.config.message}} />` ? If not, happy to help to change it to access config variables within the component's `.js` file. 

Agree that we could document how to make changes to the banner on the ember website / styleguide repos. So a rapid change to the banner is as straightforward as possible.

**Comment by amyrlam** [comment]
@fozy81 @mansona I'm not 100% sure about why it's in config/environment. Let me xpost to Discord

**Comment by amyrlam** [comment]
sorry i'm a broken record on this one, never getting to it! on the todo list for this week...

**Comment by fozy81** [comment]
@amyrlam No worries! Thanks.

**Comment by amyrlam** [comment]
will open a new branch and can cherry pick these commits, sorry my oss rate is so slow these days

**Pull request closed without merging**